spec/features/repositories/create_repository_spec.rb

```diff
@@ -0,0 +1,196 @@
+
+  let(:scm_vendor_input) { find('select[name="scm_vendor"]') }
+
+  before do
+    allow(User).to receive(:current).and_return current_user
+    allow(Setting).to receive(:enabled_scm).and_return(enabled_scms)
+
+    allow(OpenProject::Configuration).to receive(:[]).and_call_original
+    allow(OpenProject::Configuration).to receive(:[]).with('scm').and_return(config)
+  end
+
+  describe 'vendor select' do
+    before do
+      settings_page.visit_repository_settings
+    end
+    shared_examples 'shows enabled scms' do
+      it 'displays the vendor selection' do
+        expect(scm_vendor_input).not_to be_nil
+        enabled_scms.each do |scm|
+          expect(scm_vendor_input).to have_selector('option', text: scm)
+        end
+      end
+    end
+
+    context 'with the default enabled scms' do
+      it_behaves_like 'shows enabled scms'
+    end
+
+    context 'with only one enabled scm' do
+      let(:enabled_scms) { %w[Subversion] }
+      it_behaves_like 'shows enabled scms'
+      it 'does not show git' do
+        expect(scm_vendor_input).not_to have_selector('option', text: 'Git')
+      end
+    end
+  end
+
+  describe 'with submitted vendor form' do
+    let(:scm_types) { page.all('input[name="scm_type"]') }
+    before do
+      settings_page.visit_repository_settings
+      scm_vendor_input.find('option', text: vendor).select_option
+    end
+
+    shared_examples "displays only the type" do |type|
+      it 'should display one type, but expanded' do
+        expect(scm_types.length).to eq(1)
+        expect(scm_types[0].value).to eq(type)
+        expect(scm_types[0][:selected]).to be_truthy
+        expect(scm_types[0][:disabled]).to be_falsey
+
+        content = find("#toggleable-attribute-group--content-#{type}")
+        expect(content).not_to be_nil
+        expect(content[:hidden]).to be_falsey
+      end
+    end
+
+    shared_examples 'displays collapsed type' do |type|
+      let(:selector) { find("input[name='scm_type'][value='#{type}']") }
+
+      it 'should display a collapsed type' do
+        expect(selector).not_to be_nil
+        expect(selector[:selected]).to be_falsey
+        expect(selector[:disabled]).to be_falsey
+
+        content = find("#toggleable-attribute-group--content-#{type}", visible: false)
+        expect(content).not_to be_nil
+        expect(content[:hidden]).to be_truthy
+      end
+    end
+
+    shared_examples 'has managed and other type' do |type|
+      it_behaves_like 'displays collapsed type', type
+      it_behaves_like 'displays collapsed type', 'managed'
+
+      it 'can toggle between the two' do
+        find("input[name='scm_type'][value='#{type}']").set(true)
+        content = find("#toggleable-attribute-group--content-#{type}")
+        expect(content).not_to be_nil
+        expect(content[:hidden]).to be_falsey
+
+        find('input[type="radio"][value="managed"]').set(true)
+        content = find("#toggleable-attribute-group--content-managed")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.